src/NuGet.Protocol.Core.v3/HttpHandlerResourceV3Provider.cs

```diff
@@ -66,12 +64,7 @@ private HttpClientHandler TryGetCredentialAndProxy(PackageSource packageSource)
                 && !String.IsNullOrEmpty(packageSource.UserName)
                 && !String.IsNullOrEmpty(packageSource.Password))
             {
-                var cache = new CredentialCache();
```

Comment:
Looking at this again, David and I may have misunderstood something in our discussion with Yishai.  My understanding now is that CredentialPromptWebRequestHandler gets non-domain-specific credentials, which it will happily use on all requests, regardless of the request hostname.  That same CredentialPromptWebRequestHandler is re-used in multiple requests which may be for different hostnames.

If we're still misunderstanding something, please correct me, but otherwise this change looks fine to us.  We're validating it now against our scenarios.